spanner/cloud-client/src/main/java/com/example/spanner/LoadCsvExample.java

```diff
@@ -0,0 +1,250 @@
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Options;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+/** Sample showing how to load CSV file data into Spanner */
+public class LoadCsvExample {
+
+  public static final String EXCEL = "EXCEL";
+  public static final String POSTGRESQL_CSV = "POSTGRESQL_CSV";
+  public static final String POSTGRESQL_TEXT = "POSTGRESQL_TEXT}";
+
+  enum SpannerDataTypes {
+    INT64,
+    FLOAT64,
+    BOOL,
+    DATE,
+    TIMESTAMP,
+    STRING
+  }
+
+  public static ArrayList<SpannerDataTypes> spannerSchema = new ArrayList<>();
+  public static ArrayList<String> headers = new ArrayList<>();
+  public static Connection connection;
+
+  public static SpannerDataTypes parseSpannerDataType(String columnType) {
+    if (columnType.matches("STRING(?:\\((?:MAX|[0-9]+)\\))?")) {
+      return SpannerDataTypes.STRING;
+    } else if (columnType.equalsIgnoreCase("FLOAT64")) {
+      return SpannerDataTypes.FLOAT64;
+    } else if (columnType.equalsIgnoreCase("INT64")) {
+      return SpannerDataTypes.INT64;
+    } else if (columnType.equalsIgnoreCase("BOOL")) {
+      return SpannerDataTypes.BOOL;
+    } else if (columnType.equalsIgnoreCase("DATE")) {
+      return SpannerDataTypes.DATE;
+    } else if (columnType.equalsIgnoreCase("TIMESTAMP")) {
+      return SpannerDataTypes.TIMESTAMP;
+    } else {
+      throw new IllegalArgumentException(
+          "Unrecognized or unsupported column data type: " + columnType);
+    }
+  }
+
+  public static void parseTableTypes(String tableId) throws SQLException {
+    ResultSet spannerType = connection.createStatement()
+        .executeQuery("SELECT spanner_type FROM information_schema.columns "
+            + "WHERE table_name = \"" + tableId + "\"");
+    while (spannerType.next()) {
+      spannerSchema.add(parseSpannerDataType(spannerType.getString("spanner_type")));
+    }
+  }
+
+  public static void parseTableHeaders(String tableId) throws SQLException {
+    ResultSet tableHeader = connection.createStatement()
+        .executeQuery("SELECT column_name FROM information_schema.columns "
+            + "WHERE table_name = \"" + tableId + "\"");
+    while (tableHeader.next()) {
+      headers.add(tableHeader.getString("column_name"));
+    }
+  }
+
+  public static boolean isValidHeaders(CSVParser parser) {
+    List<String> allHeaders = parser.getHeaderNames();
+    for (int i = 0; i < allHeaders.size(); i++) {
+      if (!allHeaders.get(i).equals(headers.get(i))) {
+        System.err.println(
+            "Header " + allHeaders.get(i) + " does not match database table header " + headers
```

Comment:
Maybe change to "does not match database table column name" instead? A table doesn't have a header.